### ISSUE TITLE
Cross-validation for estimating generalization performance and model selection

### ISSUE THREAD

**Issue by michaelchughes** [tufts-ml/ml-research-reading-lists#10]
# Overview figure of nested cross-validation

See here: <https://sebastianraschka.com/faq/docs/evaluate-a-model.html#scenario-3>

> MCH: If you understand nested CV, you understand 80% of what is needed. If you further understand how to report *uncertainty* in nested CV results, that is a big bonus.

# Manuscripts describing practical best practices

#### Model Evaluation, Model Selection, and Algorithm Selection in Machine Learning
Sebastian Raschka
Tech Report, 2018
https://arxiv.org/pdf/1811.12808.pdf

> MCH: See especially bits about nested cross-validation

#### On Over-fitting in Model Selection and Subsequent Selection Bias in Performance Evaluation
Cawley and Talbot
Journal of Machine Learning Research
2010
http://jmlr.csail.mit.edu/papers/volume11/cawley10a/cawley10a.pdf

* See [Figure 2](http://jmlr.csail.mit.edu/papers/volume11/cawley10a/cawley10a.pdf#page=7) for clear evidence that CV can exhibit "overfitting", if you optimize the CV metric via gradient descent
* See [Figure 3](http://jmlr.csail.mit.edu/papers/volume11/cawley10a/cawley10a.pdf#page=9) for illustration that *variance* in the metric used to select hyperparameters might be important (not just whether the metric is biased or not)

#### Bias in error estimation when using cross-validation for model selection
Varma and Simon
BMC Bioinformatics, 2006
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-91

> MCH: Carefully describes nested CV and why it is necessary.
# Manuscripts describing formal results

#### No Unbiased Estimator of the Variance of K-Fold Cross-Validation
Bengio and Grandvalet
Journal of Machine Learning Research
2004
http://www.jmlr.org/papers/volume5/grandvalet04a/grandvalet04a.pdf

* Summary: There is no universal estimator (valid for all data distributions) that can help us know what the variance of our CV performance metric is in an unbiased fashion. We need to be aware that reporting CV variance likely grossly underestimates the variance due to *bias* that arises because across folds are correlated due to overlap in training/test sets.

# Also worth reading

Efron B, Tibshirani RJ: Improvements on cross-validation: The .632+ bootstrap method. J Am Stat Assoc 1997, 92: 548–560.

Fu WJ, Carroll RJ, Wang S: Estimating misclassification error with small samples via bootstrap cross-validation. Bioinformatics 2005, 21(9):1979–1986. 10.1093/bioinformatics/bti294
